From aadedb205127bfa019d1c1da9208aa41077258e9 Mon Sep 17 00:00:00 2001
From: Wei Wang <wvw@google.com>
Date: Wed, 21 Nov 2018 07:50:33 -0800
Subject: [PATCH 07/33] Allow lmkd to renice process before killing

Bug: 118468011
Bug: 121439388
Test: mem-pressure test
Change-Id: Icf387a02243af60a3bfffba912711f037669fa7f
Merged-In: Icf387a02243af60a3bfffba912711f037669fa7f
---
 prebuilts/api/28.0/public/lmkd.te | 4 ++--
 public/lmkd.te                    | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/prebuilts/api/28.0/public/lmkd.te b/prebuilts/api/28.0/public/lmkd.te
index 472946ece..5b4a235a0 100644
--- a/prebuilts/api/28.0/public/lmkd.te
+++ b/prebuilts/api/28.0/public/lmkd.te
@@ -21,8 +21,8 @@ allow lmkd system_server:file write;
 r_dir_file(lmkd, sysfs_lowmemorykiller)
 allow lmkd sysfs_lowmemorykiller:file w_file_perms;
 
-# Send kill signals
-allow lmkd appdomain:process sigkill;
+# setsched and send kill signals
+allow lmkd appdomain:process { setsched sigkill };
 
 # Clean up old cgroups
 allow lmkd cgroup:dir { remove_name rmdir };
diff --git a/public/lmkd.te b/public/lmkd.te
index 472946ece..5b4a235a0 100644
--- a/public/lmkd.te
+++ b/public/lmkd.te
@@ -21,8 +21,8 @@ allow lmkd system_server:file write;
 r_dir_file(lmkd, sysfs_lowmemorykiller)
 allow lmkd sysfs_lowmemorykiller:file w_file_perms;
 
-# Send kill signals
-allow lmkd appdomain:process sigkill;
+# setsched and send kill signals
+allow lmkd appdomain:process { setsched sigkill };
 
 # Clean up old cgroups
 allow lmkd cgroup:dir { remove_name rmdir };
-- 
2.17.1

